### PR TITLE
Update CI to go 1.11 and use go mod for vendoring

### PR DESCRIPTION
**Pull request by mavimo**
- Update CI to go 1.11
 - Move dependency to `go mod` (supported from golang 1.11)

### CODE REVIEW

**Comment by mavimo** [comment]
@aeneasr I suggest to move to go mod since can reduce "friction" for users to test it since `go mod` is the "official standard" for dependency management.

File: .travis.yml [hunk 1]
```diff
@@ -3,17 +3,18 @@ language: go
...
-before_install:
-  - curl -L -s https://github.com/golang/dep/releases/download/v${DEP_VERSION}/dep-linux-amd64 -o $GOPATH/bin/dep
-  - chmod +x $GOPATH/bin/dep
+matrix:
+   allow_failures:
+      - go: tap
 
 install:
-  - dep ensure
+  - go mod vendor
```
Comment (aeneasr) [inline .travis.yml]: No need for this

File: README.md [hunk 2]
```diff
@@ -6,11 +6,10 @@ ORY Fosite is the security first OAuth2 & OpenID Connect framework for Go. Built
 
 ## Install and run
 
-Please install [dep](https://github.com/golang/dep), then run the demo:
-
 ```
-$ go get -d github.com/ory/fosite-example
-$ cd $GOPATH/src/github.com/ory/fosite-example
-$ dep ensure
+$ git clone https://github.com/ory/fosite-example
```
Comment (aeneasr) [inline README.md]: Actually `GO111MODULE=on go get github.com/ory/fosite-example` and then running `fosite-example` should be enough

**Comment by mavimo** [comment]
@aeneasr fixes done, thx for your time to review this PR :)

**Comment by aeneasr** [comment]
Unfortunately tests fail now :(

**Comment by pmundt** [comment]
It should now also be possible to close this PR, as we've since addressed this in #17

**Comment by mavimo** [comment]
@pmundt closed :)

**Pull request closed without merging**